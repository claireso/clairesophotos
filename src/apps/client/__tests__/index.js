import React from 'react'
import { render, fireEvent } from 'react-testing-library'

import Client from '../index'

describe('<Client />', () => {
  const props = {
    translations: __TRANSLATIONS__
  }

  beforeEach(() => {
    fetch.resetMocks()
  })

  test('should render <Welcome />', () => {
    const { container } = render(<Client {...props} />)

    expect(container).toMatchSnapshot()
  })

  test('should render <Photos />', () => {
    const { container } = render(<Client {...props} {...__PHOTOS__} />)

    expect(container).toMatchSnapshot()
  })

  test('should render <Loader /> then new page', done => {
    const { container, getByTitle } = render(
      <Client {...props} {...__PHOTOS__} />
    )

    fetch.mockResponse(
      JSON.stringify({
        items: [
          {
            id: 2,
            title: 'photo 1',
            description: '',
            name: '01d3kva5d01v3074xbar7bgg0x.jpg',
            position: 'left',
            portrait: true,
            square: false,
            created_at: '2019-02-13T16:33:56.514Z',
            updated_at: '2019-02-13T16:33:56.514Z'
          }
        ],
        pager: {
          count: 2,
          totalPages: 2,
          limit: 1,
          offset: 1,
          prev: 1,
          first: 1
        }
      })
    )

    // spnapshot loader
    fireEvent.click(getByTitle('Next page'))
    expect(container).toMatchSnapshot()

    setTimeout(() => {
      expect(container.querySelectorAll('figure')).toHaveLength(1)
      done()
    }, 1000)
  })
})
