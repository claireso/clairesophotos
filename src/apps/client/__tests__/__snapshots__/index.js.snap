// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Client /> should render <Loader /> then new page 1`] = `
.c1 {
  margin: 2rem 0;
  -webkit-animation: eMLfYp 800ms ease-in-out infinite alternate;
  animation: eMLfYp 800ms ease-in-out infinite alternate;
}

.c1:after {
  background: var(--primary);
  border-radius: 50%;
  content: '';
  display: block;
  height: 1rem;
  width: 1rem;
  margin: 0 auto;
}

.c0 {
  max-width: var(--container-max-width);
  padding: 0 2rem;
  margin: 0 auto;
}

@media (min-width:800px) {
  .c0 {
    padding: 0 4rem;
  }
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    />
  </main>
</div>
`;

exports[`<Client /> should render <Photos /> 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 2rem 0;
  padding: 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 2rem 0;
  padding: 0;
}

.c2 {
  width: 100%;
  max-width: 810px;
}

.c4 {
  font-size: 1.2rem;
}

.c5 {
  color: var(--gray-2);
  display: block;
  font-size: 1rem;
  font-style: italic;
  margin: 0.4rem 0 0;
}

.c3 {
  background: var(--gray-4);
  position: relative;
  padding-top: calc(0.666667 * 100%);
  width: 100%;
  margin: 0 0 1rem;
}

.c8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 1rem 1.2rem 1.1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: white;
  border: 1px solid var(--gray-1);
  display: inline-block;
  font-family: 'Roboto',Arial,sans-serif;
  width: 4.2rem;
  color: var(--text);
  font-size: 1.6rem;
  font-weight: 700;
  will-change: border-color;
  -webkit-transition: border-color 150ms ease-out;
  transition: border-color 150ms ease-out;
}

.c8:hover {
  border-color: var(--gray-3);
}

.c7 {
  margin: 0 0 8.5rem;
  padding: 0;
  text-align: center;
}

.c7 > li {
  display: inline-block;
  margin: 0 0.5rem;
}

.c0 {
  max-width: var(--container-max-width);
  padding: 0 2rem;
  margin: 0 auto;
}

@media (min-width:800px) {
  .c1 {
    margin: 8.5rem 0;
    padding: 0 8.5rem;
  }

  .c1 + figure {
    margin-top: 20rem;
  }
}

@media (min-width:800px) {
  .c6 {
    margin: 8.5rem 0;
    padding: 0 8.5rem;
  }

  .c6 + figure {
    margin-top: 20rem;
  }
}

@media (min-width:800px) {
  .c0 {
    padding: 0 4rem;
  }
}

<div>
  <main
    class="c0"
  >
    <figure
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
        <figcaption
          class="c4"
        >
          <span />
          <span
            class="c5"
          >
            FÃ©vrier 2019
          </span>
        </figcaption>
      </div>
    </figure>
    <figure
      class="c6"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
        <figcaption
          class="c4"
        >
          <span />
          <span
            class="c5"
          >
            Janvier 2019
          </span>
        </figcaption>
      </div>
    </figure>
    <ul
      class="c7"
    >
      <li>
        <button
          class="c8"
          title="Next page"
        >
          Â»
        </button>
      </li>
      <li>
        <button
          class="c8"
          title="Last page"
        >
          Â»Â»
        </button>
      </li>
    </ul>
  </main>
</div>
`;

exports[`<Client /> should render <Welcome /> 1`] = `
.c2 {
  font-size: 3rem;
  margin: 0 0 3rem;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--primary);
}

.c3:hover,
.c3:focus {
  border-bottom: 1px solid var(--primary);
}

.c1 {
  padding: 6rem 0;
}

.c1 > p {
  font-size: 1.8rem;
}

.c0 {
  max-width: var(--container-max-width);
  padding: 0 2rem;
  margin: 0 auto;
}

@media (min-width:800px) {
  .c0 {
    padding: 0 4rem;
  }
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Welcome to your Journal ðŸŽ‰
      </h1>
      <p>
        Go to your 
        <a
          class="c3"
          href="/admin"
        >
          admin
        </a>
         to publish your first photo
      </p>
    </div>
  </main>
</div>
`;
